pkg/cmd/roachtest/roachtestutil/disk_usage.go

```diff
@@ -133,34 +134,23 @@ func GetDiskUsageInBytes(
 	ctx context.Context, c cluster.Cluster, logger *logger.Logger, nodeIdx int,
 ) (int, error) {
 	var result install.RunResultDetails
-	for {
-		var err error
-		// `du` can warn if files get removed out from under it (which
-		// happens during RocksDB compactions, for example). Discard its
-		// stderr to avoid breaking Atoi later.
-		// TODO(bdarnell): Refactor this stack to not combine stdout and
-		// stderr so we don't need to do this (and the Warning check
-		// below).
+	var err error
+	// `du` can warn if files get removed out from under it (which
+	// happens during RocksDB compactions, for example). Discard its
+	// stderr to avoid breaking Atoi later.
+	// TODO(bdarnell): Refactor this stack to not combine stdout and
+	// stderr so we don't need to do this (and the Warning check
+	// below).
+	for r := retry.StartWithCtx(ctx, *install.DefaultRetryOpt); r.Next(); {
```

Comment:
I didn't even know we have multiple versions of essentially the same operation. The test runner uses `saveDiskUsageToLogsDir` (from `collectArtifacts`), which uses the hardcoded `/mnt/data1` instead of `{store-dir}`. It seems like a further refactoring would pay dividends on having one version that's correct and reliable.

Reply:
Done. Although I had to stuff the `grep` and `>>` commands as args to the new helper, which is arguably a bit confusing that we allow the caller to do more than just "GetDiskUsage".

Reply:
> which is arguably a bit confusing that we allow the caller to do more than just "GetDiskUsage".

Agreed. We don't really have an API for composing remote commands, which could make this cleaner. But it seems ok for now, considering the callers of `GetDiskUsage` aren't likely to change.

Reply:
On second thought, `GetDiskUsage` isn't doing anything specific, really. We might as well "inline" it by replacing with a direct call to `RunWithDetails`.